### PR TITLE
Pass signals about dead child processes to GAP

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This the IO part of https://github.com/gap-system/gap/pull/1851

### CODE REVIEW

File: src/io.c [hunk 1]
```diff
@@ -154,13 +155,15 @@ RETSIGTYPE IO_SIGCHLDHandler( int whichsig )
...
-                Pr("#E Overflow in table of terminated processes\n",0,0);
+            if(!CheckChildStatusChanged(retcode, status)) {
+                if (!statsfull) {
+                    stats[lastats] = status;
+                    pids[lastats++] = retcode;
+                    if (lastats >= maxstats) lastats = 0;
+                    if (lastats == fistats) statsfull = 1;
+                } else
+                    Pr("#E Overflow in table of terminated processes\n",0,0);
+            }
```
Comment (fingolfin) [inline src/io.c]: In `FuncIO_WaitPid`, very similar code exists. Should that perhaps also call `CheckChildStatusChanged`?

File: src/io.c [hunk 2]
```diff
@@ -154,13 +155,15 @@ RETSIGTYPE IO_SIGCHLDHandler( int whichsig )
...
     if (retcode > 0) {   /* One of our child processes terminated */
         if (WIFEXITED(status) || WIFSIGNALED(status)) {
-            if (!statsfull) {
-                stats[lastats] = status;
-                pids[lastats++] = retcode;
-                if (lastats >= maxstats) lastats = 0;
-                if (lastats == fistats) statsfull = 1;
-            } else
-                Pr("#E Overflow in table of terminated processes\n",0,0);
+            if(!CheckChildStatusChanged(retcode, status)) {
```
Comment (fingolfin) [inline src/io.c]: I am somewhat unhappy that this change breaks compatibility with all prior versions of GAP (and indeed, even if the current `master` branch). I'd be more comfortable if this was conditional inside an `#ifdef`.
One way to do that would be to add a check for `CheckChildStatusChanged` to `configure`; another, perhaps more convenient (for package developers) approach would be if GAP added something like `#define GAP_HAVE_CheckChildStatusChanged 1`.

Also: missing space after `if`

File: src/io.c [hunk 3]
```diff
@@ -154,13 +155,15 @@ RETSIGTYPE IO_SIGCHLDHandler( int whichsig )
...
+                if (!statsfull) {
+                    stats[lastats] = status;
```
Comment (fingolfin) [inline src/io.c]: Perhaps avoid changing the indentation by writing this like so:
```C
#ifdef GAP_HAVE_CheckChildStatusChanged
    if (CheckChildStatusChanged(retcode, status)) {
        // GAP took care of it, nothing to do for us
    }
    else
#endif
    if (!statsfull) {
        ...
```

File: src/io.c (hunk 2 above)
Comment (ChrisJefferson) [reply src/io.c]: I tried to think of how to make this work, should have thought of just adding a `define` to GAP! WIll do that.

**Pull request merged**